File: Languages/Elixir/Lib/ListadorTarefas.ex
defmodule ListadorTarefas do
  def start do
    loop([])
  end

  defp loop(tarefas) do
    IO.puts("Escolha uma opção:")
    IO.puts("1. Adicionar tarefa")
    IO.puts("2. Listar tarefas")
    IO.puts("3. Remover tarefa")
    IO.puts("4. Sair")

    case IO.gets("> ") |> String.trim() do
      "1" -> 
        tarefas = adicionar_tarefa(tarefas)
        loop(tarefas)

      "2" -> 
        listar_tarefas(tarefas)
        loop(tarefas)

      "3" -> 
        tarefas = remover_tarefa(tarefas)
        loop(tarefas)

      "4" -> 
        IO.puts("Saindo...")
      
      _ -> 
        IO.puts("Opção inválida. Tente novamente.")
        loop(tarefas)
    end
  end

  defp adicionar_tarefa(tarefas) do
    IO.puts("Digite a tarefa:")
    tarefa = IO.gets("> ") |> String.trim()
    [tarefa | tarefas]
  end

  defp listar_tarefas(tarefas) do
    if tarefas == [] do
      IO.puts("Nenhuma tarefa encontrada.")
    else
      IO.puts("Tarefas:")
      Enum.each(tarefas, &IO.puts("- #{&1}"))
    end
  end

  defp remover_tarefa(tarefas) do
    IO.puts("Digite a tarefa a remover:")
    tarefa_remover = IO.gets("> ") |> String.trim()

    if tarefa_remover in tarefas do
      tarefas -- [tarefa_remover]
      IO.puts("Tarefa '#{tarefa_remover}' removida.")
    else
      IO.puts("Tarefa '#{tarefa_remover}' não encontrada.")
    end

    tarefas
  end
end

File: Languages/Fortan/listador_tarefas.f90
program listador_tarefas
    implicit none
    integer :: opcao, i
    character(len=100) :: tarefas(100)
    integer :: num_tarefas

    num_tarefas = 0

    print *, "Bem-vindo ao Listador de Tarefas!"

    do
        print *, "Escolha uma opção:"
        print *, "1. Adicionar tarefa"
        print *, "2. Listar tarefas"
        print *, "3. Remover tarefa"
        print *, "4. Sair"
        read(*,*) opcao

        select case (opcao)
        case (1)
            if (num_tarefas < 100) then
                print *, "Digite a tarefa:"
                read(*,'(A)') tarefas(num_tarefas + 1)
                num_tarefas = num_tarefas + 1
            else
                print *, "Limite de tarefas atingido."
            end if
        case (2)
            if (num_tarefas == 0) then
                print *, "Nenhuma tarefa encontrada."
            else
                print *, "Tarefas:"
                do i = 1, num_tarefas
                    print *, "-", tarefas(i)
                end do
            end if
        case (3)
            if (num_tarefas == 0) then
                print *, "Nenhuma tarefa para remover."
            else
                print *, "Digite o número da tarefa a remover (1 a ", num_tarefas, "):"
                integer :: tarefa_remover
                read(*,*) tarefa_remover
                if (tarefa_remover >= 1 .and. tarefa_remover <= num_tarefas) then
                    do i = tarefa_remover, num_tarefas - 1
                        tarefas(i) = tarefas(i + 1)
                    end do
                    num_tarefas = num_tarefas - 1
                    print *, "Tarefa removida."
                else
                    print *, "Número da tarefa inválido."
                end if
            end if
        case (4)
            print *, "Saindo..."
            exit
        case default
            print *, "Opção inválida. Tente novamente."
        end select
    end do
end program listador_tarefas